File: Go/ExampleTests/common/itempriceinput_string.go
package common

type ItemPriceInputString struct {
	TotalItems string
}

func NewItemPriceInputStringFromSlice(v []string) ItemPriceInputString {
	s := ItemPriceInputString{}
	if len(v) > 0 { s.TotalItems = v[0] }
	return s
}

// NewItemPriceInputStringFromText builds from the text form, e.g. Money as "25 USD".
func NewItemPriceInputStringFromText(text string) ItemPriceInputString {
	parts := RequireTokens(text, 1, "ItemPriceInput")
	return ItemPriceInputString{
		TotalItems: parts[0],
	}
}

func (s ItemPriceInputString) String() string {
	return Token(s.TotalItems)
}

func (s ItemPriceInputString) Equals(o ItemPriceInputString) bool {
	return DNCEqual(s.TotalItems, o.TotalItems)
}

func EqualItemPriceInputStringSlices(a, b []ItemPriceInputString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/itempriceinput_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct ItemPriceInputString {
    pub totalitems: String,
}

impl ItemPriceInputString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            totalitems: v.get(0).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 1, "ItemPriceInput");
        Self {
            totalitems: parts[0].clone(),
        }
    }
}

impl std::fmt::Display for ItemPriceInputString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.totalitems)].join(" "))
    }
}

impl PartialEq for ItemPriceInputString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.totalitems, &other.totalitems)
    }
}
